src/main/features/response/routes/full-admission/payment-plan.ts

```diff
@@ -1,26 +1,45 @@
 import * as express from 'express'
 
+import * as _ from 'lodash'
+
 import { Paths, FullAdmissionPaths } from 'response/paths'
 
 import { ErrorHandling } from 'shared/errorHandling'
 import { Form } from 'forms/form'
 import { DraftService } from 'services/draftService'
 import { User } from 'idam/user'
-import { PaidAmount } from 'ccj/form/models/paidAmount'
 import { DefendantPaymentPlan } from 'response/form/models/defendantPaymentPlan'
 import { FormValidator } from 'forms/validation/formValidator'
 import { FeatureToggleGuard } from 'guards/featureToggleGuard'
 import { ResponseDraft } from 'response/draft/responseDraft'
 import { Draft } from '@hmcts/draft-store-client'
 import { Claim } from 'claims/models/claim'
+import { createPaymentPlan } from 'common/paymentPlan'
+import { PaymentSchedule } from 'features/ccj/form/models/paymentSchedule'
+
+function mapFrequencyInWeeks (frequency: PaymentSchedule): number {
+  switch (frequency) {
+    case PaymentSchedule.EACH_WEEK:
+      return 1
+    case PaymentSchedule.EVERY_TWO_WEEKS:
+      return 2
+    case PaymentSchedule.EVERY_MONTH:
+      return 4
+    default:
+      return undefined
```

Comment:
Please throw an error. An undefined is not allowed value and since this is private method we control the execution paths. It should never happen and if it does we should know about it sooner then later.

Reply:
done